src/sparselm/model/_composite.py

```diff
@@ -0,0 +1,179 @@
+"""Composite estimator for piece-wise fitting."""
+
+__author__ = "Fengyu Xie"
+
+import re
+from itertools import chain
+
+import numpy as np
+from numpy.typing import ArrayLike
+from sklearn.base import RegressorMixin
+from sklearn.linear_model._base import LinearModel
+
+
+class CompositeEstimator(RegressorMixin, LinearModel):
+    """A composite estimator used to do piece-wise fitting.
+
+    The first estimator in the composite will be used to fit
+    certain features (a piece of the feature matrix, called scope)
+    to the target vector, and the residuals are fitted to the rest
+    of features by using the next estimators in the composite.
+    """
+
+    def __init__(self, estimators, estimator_scopes):
+        """Initialize estimator.
+
+        Args:
+            estimators(list[CVXEstimator]):
+                A list of CVXEstimators to add. CompositeEstimator
+                cannot be a member of CompositeEstimator.
+                An estimator will fit the residuals leftover by
+                the previous estimator fits in the list.
+            estimator_scopes(list[list[int]]):
```

Comment:
I will call it "estimator_feature_indices" since "estimator_indices" is still confusing. It could be understood as the indices of estimators in some list, etc.